protocol-units/da-sequencer/node/src/celestia/mod.rs

```diff
@@ -41,44 +45,53 @@ pub trait DaSequencerExternalDa: Clone {
 #[derive(Clone, Copy, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
 pub struct CelestiaHeight(u64);
 
+impl CelestiaHeight {
+	pub fn new(raw: u64) -> Self {
+		CelestiaHeight(raw)
+	}
+}
+
 /// Message, use to notify CelestiaClient activities.
 #[derive(Debug)]
 pub enum ExternalDaNotification {
-	/// Notify that the block at specified height has been sent to the Celestia network.
-	BlockSent(BlockHeight),
 	/// Notify that the block at specified height has been commited on celestia network
-	BlockCommited(BlockHeight, CelestiaHeight),
+	BlockCommitted(BlockHeight, CelestiaHeight),
 	/// Ask to send the block at specified height to the Celestia client.
 	/// Use during bootstrap to request block that are missing on Celestia network.
 	RequestBlock { at: BlockAt, callback: oneshot::Sender<Option<SequencerBlock>> },
 }
 
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
 pub enum BlockAt {
 	Height(BlockHeight),
-	Digest(SequencerBlockDigest),
+	Digest([u8; SequencerBlockDigest::DIGEST_SIZE]),
```

Comment:
Shall we also change the `BlockHeight` to `u64`?

Reply:
I think it's better to keep the domain type as the parameter. I would create two separate methods, unless you have reasons to commingle them with the `BlockAt` enum.

Reply:
Since @mzabaluev added a `BlockHeight` to the `SequencerBlockDigest`, which we keep in the Celestia `Blob`, the bootstrapping algo doesn't need to request blocks for a digest to find out the block height. The `enum BlockAt` can be removed.

Reply:
So you remove it in this PR or a next one?

Reply:
Heads up: I have converted this explicit array into the `block::Id` domain type.